### ISSUE TITLE
Make use of C# 7 deconstruct feature to "super charge" IEnumerable<T> cons support

### ISSUE THREAD

**Issue by DavidArno** [DavidArno/SuccincT#15]
Currently, Succinc&lt;T&gt; can be used to split an `IEnumerable<T>` into a head and tail via the following method:

````cs
var consResult = someEnumeration.ToConsEnumerable().Cons();
// consResult contains Head and Tail
````
With C# 7 deconstruction though, a simple `Deconstruct` extension method on `IEnumerable<T>` would not only allow the above to be written as:
````cs
var (head, tail) = someEnumeration;
````
But also, a deconstruction can be recursive, so to extract the first three items from a list and have `tail` represent the fourth onwards elements, the code would simply be:
````cs
var (head1, (head2, (head3, tail))) = someList;
````

Modify  Succinc&lt;T&gt; to support this.

**Comment by DavidArno** [comment]
Marking this as a breaking change as I'm removing `public static ConsResult<T> Cons<T>(this IEnumerable<T> collection)` and replacing it with the deconstruct. `ConsResult<T> Cons` will still exist for `IConsEnumerable` though.

**Comment by DavidArno** [comment]
Included in the v3.0.0 release.
